tokio/src/sync/rwlock.rs

```diff
@@ -103,6 +103,52 @@ pub struct RwLockWriteGuard<'a, T: ?Sized> {
     lock: &'a RwLock<T>,
 }
 
+impl<'a, T> RwLockWriteGuard<'a, T> {
+    /// Atomically downgrades a write lock into a read lock without allowing
+    /// any writers to take exclusive access of the lock in the meantime.
```

Comment:
Hmm, because Tokio's locks are fairly queued, if the next queued waiter is a writer, no additional waiters will be allowed access. I'm not sure if that behavior is actually an _issue_, or not, but it's worth noting.

Reply:
It's the same behavior as if the holder of the (to be downgraded) write guard had initially acquired a read guard instead. It might not always be _ideal_, but it's the behavior required to avoid starving writers. I'll add a note.